pooch/tests/test_downloaders.py

```diff
@@ -35,6 +47,38 @@ def test_ftp_downloader(ftpserver):
             check_tiny_data(outfile)
 
 
+@pytest.mark.skipif(paramiko is None, reason="requires paramiko to run SFTP")
+@pytest.mark.skipif(ON_TRAVIS, reason="SFTP is not allowed on Travis CI")
+def test_sftp_downloader():
+    "Test sftp downloader"
+    with TemporaryDirectory() as local_store:
+        downloader = SFTPDownloader(username="demo", password="password")
+        url = "sftp://test.rebex.net/pub/example/pocketftp.png"
+        outfile = os.path.join(local_store, "pocketftp.png")
+        downloader(url, outfile, None)
+        assert os.path.exists(outfile)
+
+
+@pytest.mark.skipif(paramiko is None, reason="requires paramiko to run SFTP")
+@pytest.mark.skipif(ON_TRAVIS, reason="SFTP is not allowed on Travis CI")
+def test_sftp_downloader_fail_if_fobj():
+    "Downloader should fail when a file object rather than string is passed"
+    with TemporaryDirectory() as local_store:
+        downloader = SFTPDownloader(username="demo", password="password")
+        url = "sftp://test.rebex.net/pub/example/pocketftp.png"
+        outfile = os.path.join(local_store, "pocketftp.png")
+        outfile_obj = open(outfile, "wb")
+        with pytest.raises(TypeError):
+            downloader(url, outfile_obj, None)
```

Comment:
```suggestion
        with open(outfile, "wb") as outfile_obj:
            with pytest.raises(TypeError):
                downloader(url, outfile_obj, None)
```

The Windows failures are coming from here. The temporary directory is trying to clean up while the file object is still open. Using a `with` statement should fix this.